traffic_control/clients/python/setup.py

```diff
@@ -61,6 +61,17 @@
 	           "requests>=2.13.0",
 	           "munch>=2.1.1",
 	       ],
+	       entry_points={
+	           'console_scripts': [
+	       	    'toget=to_access:get',
+	       	    'toput=to_access:put',
+	       	    'topost=to_access:post',
+	       	    'todelete=to_access:delete',
+	       	    'tooptions=to_access:options',
```

Comment:
Actually, all of the Go endpoints support options and head implicitly. I have a PR open to add an OPTIONs method endpoint - though idk if it's going anywhere. The way this is written it won't need to change at all for APIv2. It doesn't cost anything to make this forward compatible, and we can't really know whether or not we're ever "going to need it", so I don't really see the issue with including these.

Reply:
You _can_ submit a patch request to the TO server and have it fail with a catastrophically inappropriate error message. You could also try to do `toget foo` and have it fail. These scripts aren't like the client code itself, which has a specific `get_cachegroups()` method to get cachegroups for a user without them knowing anything about the request method, path, or even protocol. `to{{METHOD}}` send a `METHOD` request with no guarantees about the validity of the request, or the server's ability to respond.

Reply:
I wasn't sure so I tested it in an Ubuntu vm.

```c
int main() {
    return 256;
}
```
```console
user@host ~ $ gcc -o test test.c
user@host ~ $ ./test.c
user@host ~ $ echo "$?"
0
```
so yeah, it appears so. I wanted the error code to available to the caller - what do you suggest I do? I could divide by 10 or return 3 on client error and 4 on server error (1 and 2 already in use) or I could just leave it alone and make a note in the docs that the error code is %256. I don't really like any of them, but those are the options.

Reply:
Maybe an option to print the http status code to stdout?

Reply:
That'll come with `--response_headers`, do you think that's good enough or is there a reason someone would specifically want it to output the status code instead of doing `toget --response_headers some/path | head -n1 | cut -d ' ' -f2` ?

Reply:
Actually, now that I think about it, without an option to suppress printing the response payload, you'd need to pipe it to `head` anyway to be sure you weren't getting the payload with it.

Reply:
I guess I'm just trying to figure out what would make these convenience programs more usable than the bash wrapper around curl that this is derived from. I'd say we should just have an option to take in a format string and only output information for the given format string, but at that point we should just be using curl anyways.

Reply:
The only real differences are full HTTP stack printing ability, overrides for environment variables used in connection and authentication, built-in output beautification, better redundant path prefix handling, and better handling of client/server errors (i.e. printing all requested information as well as exiting with an error code). Tbh, that's not a huge improvement over the bash scripts, but imo the biggest improvement is that they're automatically installed with the client package, which makes them easy to get without cloning the repository.

I think I'll divide the status code by 100 such that it exits with a 4 on client error and a 5 on server error. Beyond that, you'll need to request more information to be printed to learn more. That's just what makes the most sense to me, but I'm open to suggestions.